src/mvs_eland/E3_indicator_calculation.py

```diff
@@ -153,34 +173,72 @@ def total_demand_each_sector(dict_values):
+    )
+
+    # Append total excess in electricity equivalent to kpi
+    calculate_electricity_equivalent_for_a_set_of_aggregated_values(
+        dict_values, total_excess_dict, kpi_name=TOTAL_EXCESS
+    )
+
+    return
+
+
+def calculate_electricity_equivalent_for_a_set_of_aggregated_values(
+    dict_values, dict_of_aggregated_flows, kpi_name
+):
+    r"""
+    Calculates the electricity equivalent for a dict of aggregated flows and writes it to the KPI
+
+    Parameters
+    ----------
+    dict_values: dict
+        All simulation parameters
+
+    dict_of_aggregated_flows: dict
+        Dict of aggragated flows, with keys of energy carriers.
+
+    kpi_name: str
+        Name of the KPI to write to the results
+
+    Returns
+    -------
+    Updated dict_values.
+    """
+
+    # For totalling aggregated electricity equivalents
+    total_electricity_equivalent = 0
+
     # Write total demand per energy carrier as well as its electricity equivalent to dict_values
-    total_demand_electricity_equivalent = 0
-    for energy_carrier in total_demand_dict:
-        # Define total demand of electricity carrier in original unit
+    for energy_carrier in dict_of_aggregated_flows:
+        # Define total aggregated flow of energy carrier in original unit
         dict_values[KPI][KPI_SCALARS_DICT].update(
-            {TOTAL_DEMAND + energy_carrier: total_demand_dict[energy_carrier]}
+            {kpi_name + energy_carrier: dict_of_aggregated_flows[energy_carrier]}
         )
-        # Define total energy carrier demand in electricity equivalent
+        # Define total aggregated flow of energy carrier in electricity equivalent
         dict_values[KPI][KPI_SCALARS_DICT].update(
             {
-                TOTAL_DEMAND
+                kpi_name
                 + energy_carrier
-                + SUFFIX_ELECTRICITY_EQUIVALENT: total_demand_dict[energy_carrier]
+                + SUFFIX_ELECTRICITY_EQUIVALENT: dict_of_aggregated_flows[
+                    energy_carrier
+                ]
                 * DEFAULT_WEIGHTS_ENERGY_CARRIERS[energy_carrier][VALUE]
             }
         )
-        # Add to total demand in electricity equivalent
-        total_demand_electricity_equivalent += dict_values[KPI][KPI_SCALARS_DICT][
-            TOTAL_DEMAND + energy_carrier + SUFFIX_ELECTRICITY_EQUIVALENT
+        # Add to total aggregated flow in electricity equivalent
+        total_electricity_equivalent += dict_values[KPI][KPI_SCALARS_DICT][
+            kpi_name + energy_carrier + SUFFIX_ELECTRICITY_EQUIVALENT
         ]
-    # Define total demand in electricity equivalent
+
+    # Define total aggregated flow in electricity equivalent
     dict_values[KPI][KPI_SCALARS_DICT].update(
-        {
-            TOTAL_DEMAND
-            + SUFFIX_ELECTRICITY_EQUIVALENT: total_demand_electricity_equivalent
-        }
+        {kpi_name + SUFFIX_ELECTRICITY_EQUIVALENT: total_electricity_equivalent}
     )
-    return
+    logging.info
```

Comment:
```suggestion
```
Or fill it with a message

Reply:
60917f28d016b7274eea970fc903d138178acff0